### DISCUSSION TITLE
WatchFlower v4 feedback

### DISCUSSION THREAD

**Discussion by emericg** [emericg/WatchFlower#66] [General]
In this thread you can talk about the current BETA release!
Or ask to participate! As it's currently a private beta, until we can have a couple of confirmation that the update doesn't break existing installations, or loose data.
There are a number of new features and so many improvements to existing ones, with the usual batch of newly supported sensors.

**Notable new features**
- Android background updates and notifications (experimental)
- New dynamic sensor update system, only update sensor detected near you
- New in-app sensor info screen
- New plant browser screen
- New plant selection screen
- New plant journal screen
- New nearby device scanning screen

What I'm particularly interested in is... If everything still working :)
Also are you able to enable the background updates without too much trouble.

**Comment by andrelsr** [comment]
Hi,
Background updates works, but I had to disable battery saving from killing the app. Also obviously it only works if I'm near the sensor, so many times when I open the app it hasn't been updated for some hours.
I had an issue when I left home for one week and it didn't sync the history for that week when I came back. The official flower care app didn't sync that history either.
About the journal: date picker when making a new entry selects a day before where I touch (touch day 2, it selects day 1) and sometimes it doesn't work when changing to past month, for instance. And after making the entry, touching the edit button doesn't work.

On history, for temperature and luminosity I have a solid color and a transparent one. Is that average and max values? I couldn't find the graph legend. Also, I suppose the value shown besides the title (e.g TEMPERATURE 19.7°C) refers to the average value, so I don't know what was the max value on a given day. It would be nice to have both values displayed.

If I might make another suggestion, it would be really nice to be able to change x-axis on Data graph. It's my most used feature. Maybe a pinch to zoom or perhaps somewhere to change the two weeks interval for other number of days.

Thanks for this app, it's really great!

I have a Xiaomi Mi 9 SE, Android 10, Watchflower 4.1, using a Xiaomi sensor.

**Reply by emericg** [reply]
> Background updates works, but I had to disable battery saving from killing the app

Yes you need to do that manually, depending on your phone. It sucks but... that's been a problem with Android for years and years...

> Also obviously it only works if I'm near the sensor, so many times when I open the app it hasn't been updated for some hours

You can lower the update interval in the settings, to maximize your chance to have new data, but yeah at some point the sensors need to be close to your phone...

> About the journal: date picker when making a new entry selects a day before where I touch (touch day 2, it selects day 1) and sometimes it doesn't work when changing to past month, for instance. And after making the entry, touching the edit button doesn't work.

I'll have a look.

> Is that average and max values? I couldn't find the graph legend. Also, I suppose the value shown besides the title (e.g TEMPERATURE 19.7°C) refers to the average value, so I don't know what was the max value on a given day. It would be nice to have both values displayed.

Yes it is. Good idea, I'll add it to the legend when a bar is clicked.

> If I might make another suggestion, it would be really nice to be able to change x-axis on Data graph. It's my most used feature. Maybe a pinch to zoom or perhaps somewhere to change the two weeks interval for other number of days.

Are you talking about the "all-in-one" graph from the first page or the histograms?

For the histograms you will be able to change current days/weeks/months in the next release.
For the aio graph I was planning to try a little 7/14 days toggle, but I'll have to see if it works well enough.

Thank you for your feedbacks!

**Reply by emericg** [reply]
I have implemented everything you mentioned for the next release. Thanks again for the feedbacks.

**Comment by h3rm** [comment]
bugs in linux-version (wf 4.1, Linux Mint/Cinnamon)):
-> manually refresh sensors data for all the devices per one-click most of the time don't work (would be cool to measure everything  during/after watering&feeding). In android manual refresh all devices works without problems eg. all 5mins. in linux no chance, even 30mins after the last refresh it is not possible to refresh all devices again. A manual single device-refresh is no problem and works on each click.
-> sync sensor history for all devices results in an app crash (i think one or both thermobeacon is the reason because a manual "sync history" for this device also results in an app crash) -> https://i.imgur.com/M8MBKwH.png or(?) https://i.imgur.com/KZd4gId.png .. i can help you debug if you tell me how...
[Update:] the appcrash also occurs when syncing history of a single sensor (both flowercare + thermobeacon)
[Update:] same Problem in WF 4.0 😢 - in WF 3.0 & 3.1. syncing history works)

what would be cool:
-> in the device map please also show the given names of each sensor. will make it easier :)
-> manuall adding (currently missing) "plants" to the database?
-> extra journal or journal grouping for all plants in the same "location"?
-> calendar in the main view for quick entries & accessibility & overview for eg. editing more sensors/plants with less clicks? :)

i just reinstalled the sensors today so i will update the other things like journal asap

**Reply by h3rm** [reply]
@emericg sorry that i didn't gave an update, but i downgraded to wf 3.1 cause i need the manual refresh function once a week :(

**Reply by emericg** [reply]
Sorry I never responded, I was working on other stuff. I'm now planning a release before the end of the year.


> manually refresh sensors data for all

I'm having a lot of trouble myself with Bluetooth on Linux, I'm not entirely sure why.
Does this problem went away when you went back to the v3.1?

> sync sensor history for all devices results in an app crash

So unfortunately I don't know what cause the "sync all" to crash, I never witness this issue on any of my devices (and I don't recall any work on the feature that could be involved).

I tried to disable the "screensaver off" feature wile syncing, that is one of the thing that *could* cause a crash? (and one thing that is unique to the sync feature). If you can try one of the recent build on linux that should be interesting.

> in the device map please also show the given names of each sensor. will make it easier :)

I'll have a look, depending on how much work it represents. The device view is somehow independent of the rest of the app, I'm not sure if I want to change that.

> manuall adding (currently missing) "plants" to the database?

That should be available "soon", but not in the app. I will publish the database in a github repo, and adding/editing plants should be as simple as adding a new xml file to a folder.

> extra journal or journal grouping for all plants in the same "location"?
> calendar in the main view for quick entries & accessibility & overview for eg. editing more sensors/plants with less clicks? :)

This could be interesting to do, but it's a lot of UI and UX work, and I'm not really planning to put much more into this feature. I've fixed a few bugs and would be open to add a few things, but to an extent :)

**Reply by h3rm** [reply]
thx for you reply! at the moment i don't use the sensors... maybe for the next time a better solution would be a "working" 24/7 android-vnc-server-app so i don't have to use linux or windows 24/7 (android 12 works perfect but vnc-server or any other remote-control-apps are crap...) .. mhmm.. i already tried https://play.google.com/store/apps/details?id=de.abr.android.avnc&hl=de_AT&gl=US (pro version) but the limitations an android versions <10 are horrible

> I'm having a lot of trouble myself with Bluetooth on Linux, I'm not entirely sure why. Does this problem went away when you went back to the v3.1?

On 3.1. it is possible to sync all sensors without any problems. I already tried about 5-6 different bluetooth-usb-adapters and i am sure that the problem is not one of the adapters :/ But i agree - linux & bluetooth is not the best combination.. it was a longer way until i found the "best bt-adapter"...

> I tried to disable the "screensaver off" feature wile syncing, that is one of the thing that _could_ cause a crash? (and one thing that is unique to the sync feature). If you can try one of the recent build on linux that should be interesting.

the notebook was running 24/7 but with "only" the screensaver "on" so that the dispay don't "burn" when using it 24/7.. i will try to disable it but i don't think that this is the solution cause the problem also occurs instantly after starting the device, opening the WF-app and "sync all sensors". but i will update in a few weeks!

> > in the device map please also show the given names of each sensor. will make it easier :)
> I'll have a look, depending on how much work it represents. The device view is somehow independent of the rest of the app, I'm not sure if I want to change that.

don't worry, it's not so important! :)

> > manuall adding (currently missing) "plants" to the database?
> That should be available "soon", but not in the app. I will publish the database in a github repo, and adding/editing plants should be as simple as adding a new xml file to a folder.

yay :)


> > extra journal or journal grouping for all plants in the same "location"?
> > calendar in the main view for quick entries & accessibility & overview for eg. editing more sensors/plants with less clicks? :)
> This could be interesting to do, but it's a lot of UI and UX work, and I'm not really planning to put much more into this feature. I've fixed a few bugs and would be open to add a few things, but to an extent :)

also don't worry. it is not that necessary or a killer feature that everyone "needs" :)

thx!
